analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Invocation.cs

```diff
@@ -413,5 +413,123 @@ public void Invocation_IsNullOrEmpty_TryFinally()
                 new SymbolicValue().WithConstraint(ObjectConstraint.NotNull)
             });
         }
+
+        [DataTestMethod]
+        [DynamicData(nameof(ThrowHelperCalls))]
+        public void Invocation_ThrowHelper_StopProcessing(string throwHelperCall)
+        {
+            var code = $@"
+Tag(""Before"");
+{throwHelperCall}
+Tag(""Unreachable"");
+";
+            var validator = SETestContext.CreateCS(code).Validator;
+            validator.ValidateTagOrder("Before");
+            validator.ValidateExitReachCount(0);
+            validator.ValidateExecutionCompleted();
+        }
+
+        [DataTestMethod]
+        [DynamicData(nameof(ThrowHelperCalls))]
+        public void Invocation_ThrowHelper_OnlyInBranch(string throwHelperCall)
+        {
+            var code = @$"
+if (condition)
+{{
+    Tag(""Before"");
+    {throwHelperCall}
+    Tag(""Unreachable"");
+}}
+Tag(""End"");
+";
+            var validator = SETestContext.CreateCS(code, ", bool condition").Validator;
+            validator.ValidateTagOrder("Before", "End");
+            validator.ValidateExitReachCount(1);
+            validator.ValidateExecutionCompleted();
+        }
+
+        [DataTestMethod]
+        [DynamicData(nameof(ThrowHelperCalls))]
+        public void Invocation_ThrowHelper_TryCatchFinally(string throwHelperCall)
+        {
+            var code = @$"
+try
+{{
+    {throwHelperCall}
+    Tag(""Unreachable"");
+}}
+catch
+{{
+    Tag(""Catch"");
+}}
+finally
+{{
+    Tag(""Finally"");
+}}
+Tag(""End"");
+";
+            var validator = SETestContext.CreateCS(code, ", bool condition").Validator;
+            validator.ValidateTagOrder("Catch", "Finally", "Finally", "End");
+            validator.ValidateExitReachCount(2);
+            validator.ValidateExecutionCompleted();
+        }
+
+        [DataTestMethod]
+        [DataRow("System.Diagnostics.CodeAnalysis.DoesNotReturn")]
+        [DataRow("JetBrains.Annotations.TerminatesProgram")]
+        public void Invocation_ThrowHelper_Attributes(string throwHelperAttribute)
+        {
+            var code = $@"
+using System;
+using System.Diagnostics;
+
+public class Sample
+{{
+    public void Test()
+    {{
+        Tag(""Before"");
+        ThrowHelper();
+        Tag(""Unreachable"");
+    }}
+
+    [{throwHelperAttribute}]
+    public void ThrowHelper()
+    {{
+        // No implementation. The attribute should drive the analysis.
+    }}
+
+    static void Tag(string name) {{ }}
+}}
+
+namespace JetBrains.Annotations
+{{
+    [Obsolete(""Use [ContractAnnotation('=> halt')] instead"")]
+    [AttributeUsage(AttributeTargets.Method)]
+    public sealed class TerminatesProgramAttribute : Attribute {{ }}
```

Comment:
copied from [fuget](https://www.fuget.org/packages/JetBrains.Annotations/2022.1.0/lib/netstandard2.0/JetBrains.Annotations.dll/JetBrains.Annotations/TerminatesProgramAttribute?code=true)

`[ContractAnnotation]` support is also fairly easy to do here and we should consider it for the future.